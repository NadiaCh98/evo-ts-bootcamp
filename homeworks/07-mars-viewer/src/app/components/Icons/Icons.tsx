export { ReactComponent as FavIcon } from '../../../assets/icons/fav.svg';
export { ReactComponent as RemoveIcon } from '../../../assets/icons/remove.svg';
