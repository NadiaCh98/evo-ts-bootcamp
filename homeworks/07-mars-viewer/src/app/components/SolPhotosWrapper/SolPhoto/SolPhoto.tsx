import React, { ReactElement } from 'react';
import styles from './SolPhoto.module.css';

interface SolPhotoProps {
  readonly photoSrc: string;
  readonly title: string;
  readonly icon: ReactElement;
}

const SolPhoto: React.FC<SolPhotoProps> = ({
  photoSrc,
  title,
  icon,
}: SolPhotoProps) => {
  return (
    <div className={styles.SolPhoto}>
      {icon}
      <img src={photoSrc} alt={photoSrc} />
      <p>{title}</p>
    </div>
  );
};

export default SolPhoto;
