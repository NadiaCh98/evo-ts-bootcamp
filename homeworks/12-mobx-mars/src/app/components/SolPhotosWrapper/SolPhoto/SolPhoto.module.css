.SolPhoto {
    min-height: 100px;
    position: relative;
    display: flex;
    flex-flow: column;
}

.SolPhoto img {
    width: 100%;
    height: 100%;
    object-fit: cover;
}

.SolPhoto p {
    background: var(--background-color);
    color: var(--paragraph-color);
    text-align: center;
    font-size: 0.8rem;
    line-height: 1.2rem;
}

.SolPhoto svg {
    width: 2rem;
    position: absolute;
    top: var(--grid-gap);
    left: var(--grid-gap);
    transition: width 0.5s;
}

.SolPhoto svg:hover {
    cursor: pointer;
    width: 3rem;
}