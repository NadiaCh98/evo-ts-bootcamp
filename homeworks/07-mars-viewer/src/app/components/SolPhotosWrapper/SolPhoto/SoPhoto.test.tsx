import React from 'react';
import SolPhoto from './SolPhoto';
import { render } from '@testing-library/react';
import { FavIcon } from '../../Icons/Icons';

describe('SolPhoto Component', () => {
  const imageSrc =
    'https://mars.nasa.gov/msl-raw-images/msss/00001/mhli/0001MH0000002000I1_DXXX.jpg';

  it('has a correct template #1', () => {
    const { container } = render(
      <SolPhoto photoSrc={imageSrc} title="Nasa Photo" icon={<FavIcon />} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
