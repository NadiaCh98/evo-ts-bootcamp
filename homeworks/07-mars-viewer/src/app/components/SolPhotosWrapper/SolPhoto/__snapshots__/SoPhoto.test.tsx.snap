// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SolPhoto Component has a correct template #1 1`] = `
<div
  class="SolPhoto"
>
  <svg>
    fav.svg
  </svg>
  <img
    alt="https://mars.nasa.gov/msl-raw-images/msss/00001/mhli/0001MH0000002000I1_DXXX.jpg"
    src="https://mars.nasa.gov/msl-raw-images/msss/00001/mhli/0001MH0000002000I1_DXXX.jpg"
  />
  <p>
    Nasa Photo
  </p>
</div>
`;
